src/middleware/auth.js

```diff
@@ -9,9 +10,12 @@ export const protect = asyncHandler(async (req, res) => {
       : null
     : null;
   if (!token) return makeResponse({ res, status: 403, message: 'Unauthorized' });
+  const isBackListedToken = isBlacklistedToken(token);
+  if (isBackListedToken) return makeResponse({ res, status: 403, message: 'Unauthorized' });
   const decodedUser = decodeJwtToken(token).data;
   const user = decodedUser ? await getOneUser({ _id: decodedUser._id }, false) : null;
   if (!user) return makeResponse({ res, status: 403, message: 'Unauthorized' });
+  user.user_token = token;
```

Comment:
The same old issue's present, would be better if we don't attach this to the user object, we could just do `req.user_token = token` and update its references